### PR TITLE
fix(replay): SR Breadcrumbs include level name as a string

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :scroll: Description

This PR fixes SR Breadcrumb Level serialization. The level is now included as a string, which is used in the UI. Android and Browser work the same.

## :bulb: Motivation and Context

Before the levels were send as numbers.

## :green_heart: How did you test it?

unit test, sample app

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/include/SentryLevelHelper.h
Comment (bruno-garcia) [inline Sources/Sentry/include/SentryLevelHelper.h:12]: This is a breaking change as it's a public API.

I think it's best we add a new method. And use that internally instead. And we can annotate the old one as obsolete

**Review by bruno-garcia — CHANGES_REQUESTED** [review]
We need to keep the current public API unchanged

File: Sources/Sentry/include/SentryLevelHelper.h
Comment (krystofwoldrich) [reply Sources/Sentry/include/SentryLevelHelper.h:12]: Make sense, I'll update it.

Comment (armcknight) [reply Sources/Sentry/include/SentryLevelHelper.h:12]: I believe you're correct, it's not public. Either way, I would prefer if instead of modifying the signature, nameForSentryLevel were used inside SentryRRWebBreadcrumbEvent.init.

Comment (krystofwoldrich) [reply Sources/Sentry/include/SentryLevelHelper.h]: Thanks for the suggestion, I've reverted the `breadcrumbLevel` changes and added the conversion to `SentryRRWebBreadcrumbEvent`.

[`c7e4835` (#4141)](https://github.com/getsentry/sentry-cocoa/pull/4141/commits/c7e483514e1d0e467e6c810de39f4a6bf75efe1d)

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
Hm looks like requires code owner approval. Andrew is off the due to 4th July so either I force merge or we wait until monday

**Pull request merged**
